lib/rcon/packet.rb

```diff
@@ -41,10 +41,16 @@ class Packet
     def self.read_from_socket_wrapper(socket_wrapper)
       if socket_wrapper.ready_to_read?
         size = socket_wrapper.recv(INT_BYTE_SIZE).unpack(INTEGER_PACK_DIRECTIVE).first
```

Comment:
if `recv` is returning `nil` then the call to `unpack` will fail here and we'll get the nil method error (so your check below won't get hit)

wondering if if we should wrap the `recv` method in `SocketWrapper` to raise the error instead.

Reply:
Sorry, `#recv` I think is returning "" instantly when the connection is closed, which results in `size` being set to `nil`, and makes the `-` operator fail.